rfcs/0005-pablo-distribution.md

```diff
@@ -0,0 +1,667 @@
+Pablo pools. It is equal to the protocol fee charged on Pablo pools.
+This can be accomplished by calling the already existing
+`StakingReward.transfer_reward` interface as follows. According to
+product there is also a need to convert whatever the fee asset in to
+PBLO to create a demand/additional value for PBLO.
+
+<img src="0005-pablo-distribution-assets/images/images/pablo-fNFT-pblo-staking-fee-distro.png" width="510" height="346" alt="pablo fNFT pblo staking fee distro" />
+
+Will it need a change in
+[this](https://github.com/ComposableFi/composable/blob/main/frame/composable-traits/src/staking_rewards.rs#L96)
+?
+
+### 5.3. Pallet Staking Rewards - LP/PICA/PBLO/Other Token Staking Reward Pools
+
+This section covers how the staking rewards are distributed using the
+[staking rewards
+pallet](https://github.com/ComposableFi/composable/tree/main/frame/staking-rewards).
+
+#### 5.3.1. Analysis of Reward Calculations
+
+In order to create the necessary reward pool as well as the rewarding
+rate for stakers the following model can be used. It tries to address
+the following constraints (TODO move to requirements)
+
+1.  Allow <span id="rate">specification of the reward rate for a
+    pool</span> (even setting a dynamically changing rate)
+
+2.  Allow addition of new stakers at anytime to a pool, start earning
+    immediate rewards
+
+3.  Allow more realtime calculation of rewards on-demand for a given
+    pool for a given user.
+
+4.  Allow shorter reward pool calculation epoch with the use of the
+    reward rate.
+
+5.  Allow expansion of rewards pools realtime.
+
+6.  Allow extending of staked position in time and amount.
+
+7.  Allow splitting of staked position into smaller positions.
+
+8.  \[Postponed\] Allow compounding of staked position when the rewarded
+    asset is the same as staked. Not handled at the moment. Though it is
+    possible for users to just re-stake their earned assets.
+
+To analyze the requirement fully, let’s define the following terms for a
+given staking reward pool,
+
+Pre-defined reward rate (say per second) <img src="0005-pablo-distribution-assets/images/stem-6fb32a8803a6d58cd54908033a2556f9.png" width="23" height="6" alt="stem 6fb32a8803a6d58cd54908033a2556f9" /></span>
+
+Pre-defined reward calculation epoch in seconds <img src="0005-pablo-distribution-assets/images/stem-6184b58307a1dc90934a6a7051a42ceb.png" width="22" height="8" alt="stem 6184b58307a1dc90934a6a7051a42ceb" /></span>
+
+Reward per calculation epoch <img src="0005-pablo-distribution-assets/images/stem-b219ff7e7a0df744c99c2e11229a1ded.png" width="33" height="8" alt="stem b219ff7e7a0df744c99c2e11229a1ded" /></span>
+
+Previous total reward pool before the current epoch <img src="0005-pablo-distribution-assets/images/stem-53fadade13e71b863963af9a23b28b71.png" width="25" height="8" alt="stem 53fadade13e71b863963af9a23b28b71" /></span>
+
+Assuming there is a per epoch calculation which adds to the pool, the
```

Comment:
there is no need for epoches. rate is per second.

Reply:
There has to be a batch process that calculates the reward for the pool so epoch is how often that batch job runs